liblwgeom/lwgeom_topo.c

```diff
@@ -3003,7 +3003,7 @@ _lwt_FindNextRingEdge(const POINTARRAY *ring, int from,
       LWDEBUGF(1, "First point of edge %" LWTFMT_ELEMID
                   " matches ring vertex %d", isoe->edge_id, from);
       /* first point matches, let's check next non-equal one */
-      for ( j=1; j<epa->npoints; ++j )
+      for ( j=1; j<(int32_t)epa->npoints; ++j )
```

Comment:
Switching j to int64_t to avoid issues with npoints > INT32_MAX. As stated in the other comment the function needs to be reworked to guarantee no undefined behaviour when using uints.

Reply:
Can j be just uint32_t with a couple of asserts that disallow going below zero?